### PR TITLE
[ServiceBus] Stress tests refactoring and chaos

### PR DESCRIPTION
**Pull request by lmolkova**
- adds rate-limiting to send tests
- rewrites them to run a limited amount of time
- removes custom metrics in favor of ones reported by SDK
- adds chaos configuration
- adds portal workbooks to analyze test  results
- minor doc fixes in core and sb

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageReceiverAsync.java [hunk 1]
```diff
@@ -4,63 +4,40 @@
...
+        ServiceBusReceiverAsyncClient client = getReceiverBuilder(options, false).buildAsyncClient();
 
         client.receiveMessages()
             .flatMap(message -> {
                 LOGGER.verbose("message received: {}", message.getMessageId());
-                rateMeter.add(receiveCounterKey, 1);
                 return client.complete(message)
                     .onErrorResume(error -> {
-                        LOGGER.error("error happened: {}", error.getMessage());
+                        LOGGER.error("error completing", error);
```
Comment (alzimmermsft) [inline sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageReceiverAsync.java:32]: Should make sure the sync and async logic is the same. This has a wider catching of exception types than the sync case, we should restrict this down to `ServiceBusException` and `AmqpException`.

File: sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageReceiverAsync.java [hunk 2]
```diff
@@ -4,63 +4,40 @@
...
                     });
             })
-            .subscribe(
-                data -> {
-                },
-                error -> {
-                    LOGGER.error("error happened: {}", error.getMessage());
-                    telemetryClient.trackException((Exception) error);
-                });
+            .take(durationInMinutes)
```
Comment (alzimmermsft) [inline sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageReceiverAsync.java:36]: Does this not busy loop too if `count == 0`?

File: sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageSender.java [hunk 3]
```diff
@@ -23,48 +22,33 @@
...
 
-    @Value("${SEND_MESSAGES:10}")
-    private int messagesToSend;
+    @Value("${BATCH_SIZE:2}")
+    private int batchSize;
 
-    @Value("${PAYLOAD_SIZE_IN_BYTE:8}")
-    private int payloadSize;
+    @Value("${MESSAGE_SIZE_IN_BYTES:8}")
+    private int messageSize;
```
Comment (alzimmermsft) [inline sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageSender.java:35]: Should this test support concurrency similarly to the async scenario by using a ExecutorService or ForkJoinPool?

Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageSender.java:39]: it probably should, but I'd prefer to do it separately when focusing on sync concurrency validation.

File: sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageReceiverAsync.java (hunk 2 above)
Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/MessageReceiverAsync.java:36]: the receiver will push messages to the subscriber once they become available, there is no loop from the user. 
It's a bit different in the case of sync client though.

File: sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/TestUtils.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+final class TestUtils {
+    private static final Random RANDOM = new Random();
+    private static final AmqpRetryOptions RETRY_OPTIONS = new AmqpRetryOptions().setTryTimeout(Duration.ofSeconds(5));
+    private static final ClientLogger LOGGER = new ClientLogger(TestUtils.class);
+    static ServiceBusClientBuilder.ServiceBusSenderClientBuilder getSenderBuilder(ScenarioOptions options, boolean session) {
```
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/TestUtils.java:25]: nit: missing a new line before the method declaration.

File: sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/RateLimiter.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                return false;
+            }
+        } while (!bucket.compareAndSet(remaining, remaining - 1));
+
+        return true;
+    }
+
+    public void release() {
+        inFlight.decrementAndGet();
+    }
```
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/RateLimiter.java:64]: Do we need a dispose/close method for our RateLimiter type to release the `replenishTimer`? I was thinking if we need to dipose it at the end of the test, say after closing the client.

**Review by anuchandy — APPROVED** [review]
thank you! the workbooks that you created are super useful. 👍

File: sdk/servicebus/azure-messaging-servicebus-stress/portal_workbooks/run_details.json [hunk 6]
```diff
@@ -0,0 +1,210 @@
+{
+  "version": "Notebook/1.0",
+  "items": [
+    {
+      "type": 1,
+      "content": {
+        "json": "## ServiceBus for Java stress testing\n\nEnter the test runId - it matches ServiceBus resource name and is a suffix in AppInsights role instance."
```
[the remark this answers is not in the document]
Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus-stress/portal_workbooks/run_details.json]: it's ApplicationInsights Role **Instance**  - the unique identifier of a service instance (we might change it in the future once we start running multi-consumer tests)

File: sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/RateLimiter.java (hunk 5 above)
Comment (lmolkova) [reply sdk/servicebus/azure-messaging-servicebus-stress/src/main/java/com/azure/messaging/servicebus/stress/scenarios/RateLimiter.java:64]: great point, done

**Pull request merged**